### DISCUSSION TITLE
Message not encrypted

### DISCUSSION THREAD

**Discussion by masihfathi** [postalserver/postal#2506] [Help with using Postal]
Hi 
On the new installation of postal i got encryption error when trying to send email. see this gmail log
I check things that i assumed it might be related 
ssl configured correctly also checked timezone of the server.
i installed the docker version of postal.
<img width="281" alt="Screenshot 1402-03-27 at 17 12 39" src="https://github.com/postalserver/postal/assets/13271510/b25f2844-548e-446b-a3dc-b4e866aeab75">

**Comment by sinnbeck** [comment]
Can you give the steps you took to encrypt it? Maybe someone can recreate the issue and help with a solution

**Reply by masihfathi** [reply]
install ssl with caddy default configuration but it raise encryption problem for messages but domain have ssl

**Reply by sinnbeck** [reply]
If you plan to use the certificate from Caddy for the smtp server, you need to change the type for it to work. Add this in your Caddyfile at the top

```
{
  key_type rsa4096
}

**Reply by masihfathi** [reply]
i added but not work also this way need to generate standalone ssl.

**Reply by sinnbeck** [reply]
I just have a cron job that copies over the new certificates every weekend (replace [domain] with the actual domain name)

```
cp /opt/postal/caddy-data/caddy/certificates/acme-v02.api.letsencrypt.org-directory/[domain]/[domain].crt /opt/postal/config/smtp.crt
cp /opt/postal/caddy-data/caddy/certificates/acme-v02.api.letsencrypt.org-directory/[domain]/[domain].key /opt/postal/config/smtp.key
chmod o+r /opt/postal/config/smtp.*

**Reply by masihfathi** [reply]
Can postal read the generated caddy files?
i experienced later that generated caddy files for starttls in smtp not worked so i used generate ssl with lets encrypt standalone.

**Reply by sinnbeck** [reply]
It isnt. But it can read a certificate file which is what caddy is generating, and my example just copies those generated files to the specified path which is where postal will look by default

**Reply by masihfathi** [reply]
Thanks 
i will check but with pem files and also with crt file and test it

**Reply by masihfathi** [reply]
i test crt files but the problem exists

**Reply by sinnbeck** [reply]
I dont know what that means. You copied over both files and restarted postal? And then did what exactly? I dont think I have seen the error in your image when I set up my server. Can you explain what you do exactly?

**Reply by masihfathi** [reply]
Yes i copied the crt files to the config path then assign them to postal yml tls paths, then restarted postal again but when sending message i got "did not encrypt this message" error in the gmail headers.

**Reply by sinnbeck** [reply]
Ok so you are using the "Messages->Send Message" directly in the postal website? I am just trying to recreate it and cannot

**Reply by sinnbeck** [reply]
And I assume you mean both crt and key file?

**Reply by masihfathi** [reply]
All sent emails through postals not encrypted
![image](https://github.com/postalserver/postal/assets/13271510/2b1db534-e9fb-4dbc-b6f8-6fcc7e3e4964)
![image (1)](https://github.com/postalserver/postal/assets/13271510/a36fcae8-6575-471d-a4c0-aa0e59ab171d)

**Reply by sinnbeck** [reply]
Did you remember to restart caddy to obtain the other kind of certificates (after the first change)? Also no need to specify the paths in postal config as they are the default

**Reply by masihfathi** [reply]
Yes I restarted caddy docker image. The strange things here is. it doesn't have problem on an other server but in some servers it has problem in encryption. Do you think it might be related to server configs?

**Reply by sinnbeck** [reply]
Really hard to guess. But if you have a working server, comparing the exact files and config should be easy?

**Reply by masihfathi** [reply]
Everything on the postal.yml and the application side is same, cause of that I'v been confused

**Reply by sinnbeck** [reply]
Weird that you got it working on the other server without a certificate then. Anyways I don't know how to help you. If the servers are identical and same postal version, I just don't know

**Reply by masihfathi** [reply]
Thanks for your patience and your help

**Comment by willpower232** [comment]
You need to provide the SSL certificate to Postal via the config file and enable it

https://github.com/postalserver/postal/blob/8b525d0381a9e0113af808b9ec2eb47bf78ec60b/config/postal.defaults.yml#L69-L71
